test/extended/builds/gitauth.go

```diff
@@ -59,7 +59,9 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] can use private repositor
 			g.BeforeEach(func() {
 				ctx := context.Background()
 				httpToken, err := oc.AsAdmin().KubeClient().CoreV1().Secrets("build-e2e-github-secrets").Get(ctx, "github-http-token", metav1.GetOptions{})
-				o.Expect(err).NotTo(o.HaveOccurred())
+				if err != nil {
+					g.Skip("test requires secret only available in CI step registry")
+				}
```

Comment:
Check that the error is a 404/not found.

```suggestion
				if errors.IsNotFound(err) {
					g.Skip("required secret build-e2e-github-secrets/github-http-token is missing")
				}
```